### PR TITLE
macOS: send report id conditionally for hid_get_feature_report

### PR DESCRIPTION
**Pull request by Youw**
same as in hid_send_feature_report, and Windows/Linux

### CODE REVIEW

File: mac/hid.c [hunk 1]
```diff
@@ -763,38 +763,61 @@ hid_device * HID_API_EXPORT hid_open_path(const char *path)
...
 	}
-	else {
-		/* Using numbered Reports.
-		   Send the Report Number */
-		data_to_send = data;
-		length_to_send = length;
-	}
 
+	/* Avoid crash if the device has been unplugged. */
 	if (!dev->disconnected) {
```
Comment (vitalys) [inline mac/hid.c]: Probably it has to be `dev->disconnected`

**Comment by Youw** [comment]
This will likely reference https://github.com/libusb/hidapi/issues/60.

@DanielVanNoord, @micolous, please try if this works with your device(s).
It does with mine.

File: mac/hid.c (hunk 1 above)
Comment (Youw) [reply mac/hid.c]: thanks)

**Comment by Youw** [comment]
@DanielVanNoord, @micolous ping

hidapi team, your opinion/feedback is needed
with all my local tests, this implementation looks best and consistent with other systems

@z3ntu, you have originally propagated #3, so you feedback is appreciated too

**Comment by z3ntu** [comment]
Apparently the devices I use hidapi with, don't use numbered reports. I'll test this PR tomorrow if it works with my application!

**Comment by Youw** [comment]
Thanks.
The device I use, does use numbered reports, and #3 did broke my app (as I mentioned earlier, I didn't notice that, since I have my own fork with my own patches).

If this works for you, I think we will accept it this was.

**Comment by z3ntu** [comment]
Works fine for my use case :+1:

**Comment by Youw** [comment]
@Qbicz, if you don't have comments regarding the implementation, I'd like to merge this ASAP

**Comment by DanielVanNoord** [comment]
@Youw I'll grab a Mac and test it with my devices that use report ids today. Will let you know the results as soon as I can.

**Comment by Qbicz** [comment]
@youw it looks good to me. I don't have a Mac setup so I didn't notice the problem before.

**Comment by Youw** [comment]
meanwhile found bug in my own implementation
nothing conceptually changes though

**Comment by DanielVanNoord** [comment]
Ran some quick tests on a Mac with a device that uses feature report ids and everything seems to be working.

**Comment by Youw** [comment]
great, we have a winner here

**Pull request merged**